v2/namer/namer_test.go

```diff
@@ -103,3 +103,32 @@ func TestNameStrategy(t *testing.T) {
 		t.Errorf("Wanted %#v, got %#v", e, a)
 	}
 }
+
+// NOTE: this test is intended to demostrate specific behavior
+// described in https://github.com/kubernetes/kubernetes/issues/124192
+// and should not be considered a good usage scenario of Namer
```

Comment:
This is making me wonder if we could fix the callsite(s) of `rawNamer.Name` instead of the function itself. Have we located the callsite(s)? Is there one or are there many?